### PR TITLE
Node 12, xxhash v0.6.5, updated CI configs

### PR DESCRIPTION
**Pull request by ben-page**
This fixes support for NodeJS 12, updated the xxhash dependency to v0.6.5, and fixes the broken CI configs.

Also, sorry about the noise on the previous PRs. I wanted to separate these three items into separate PRs. There was just no good way to do that until the CI scripts are fixed.

The current travis config always fails for travis and doesn't build against recent versions of NodeJS. So I needed the improved CI configs to prove the Node 12 support worked. So I decided that putting them into one PR with three commits is the best choice. Let me know if you prefer something else.

### CODE REVIEW

**Comment by ben-page** [comment]
@mscdex, I'm hoping you can pull this soon. The current release does not work with Node.js 12

**Comment by mscdex** [comment]
There's a lot of extra changes and additions here that I don't really agree with. I think the changes should strictly be related to upgrading the code base to use `nan`-related functionality where possible and updating the upstream dependency (still as separate commits).

File: src/common.hpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 #define XXH_STATIC_LINKING_ONLY
 #define XXH_PRIVATE_API
-#define XXH_ACCEPT_NULL_INPUT_POINTER
+#define XXH_ACCEPT_NULL_INPUT_POINTER 1
```
Comment (ben-page) [inline src/common.hpp]: This is the only change in this commit that isn't strictly copy and pasting files.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/node_modules
```
Comment (ben-page) [inline .gitignore]: I assumed this .gitignore would make future PRs and publishing a little easier as you wont have to be sure to clear these directories by hand.

File: package.json [hunk 3]
```diff
@@ -1,17 +1,33 @@
-{ "name": "xxhash",
+{
```
Comment (ben-page) [inline package.json]: The formatting change was done by `npm i nan@latest`

**Comment by ben-page** [comment]
My intention definitely wasn't to make any unnecessary changes or addition. I'd like this to be very easy to pull. Would you mind clarifying what you object to?

I'll go over each commit to justify the changes that I made.

**upgrades xxhash to v0.6.5  (d59891f)**
The only changes I made were literally extracting the official xxHash-0.6.5.zip file into the deps folder and changing a preprocess directive to be compatible with xxhash code.

**fixes NodeJS 12 compatibility  (1604274)**
I'm not experienced with NAN, so I literally only made changes remove the errors and warnings thrown when building against Node.js 12. 
Assuming that isn't the problem. Maybe you are referring to the formatting change to the package.json file? That was done by `npm i nan@lastest`. I did not edit the file manually.
Also, I included a .gitignore file that ensures neither the node_modules or build folders are include in git or npm when publish. I thought this was a reasonable change, but I will happily remove it if you would like.

**updated CI based on the recommendations   (a8c656c)**
This updates the travis config (which is currently broken) and the appveyor config (which out of date). I started with the recommended configs from https://github.com/nodejs/node-addon-api/. The only changes I made was to add Node.js 12 (they don't include it yet) and remove chakracore.

**Comment by mscdex** [comment]
I'm not against dropping v0.10 and v0.12 support.

As far as the CI configs go, you can probably use the configs from [`mmmagic`](https://github.com/mscdex/mmmagic) as a guide since I recently updated them.

**Comment by ben-page** [comment]
OK, I imported the CI configs from mmagic. No changes were needed.

File: .gitignore (hunk 2 above)
Comment (mscdex) [reply .gitignore]: I always publish from github tarballs because of prior npm bugs, so I'm guaranteed to publish exactly what I expect.

**Comment by ben-page** [comment]
Removed .gitignore and formatting change to package.json

File: src/hash_64.hpp [hunk 4]
```diff
@@ -173,9 +175,9 @@ class Hash64 : public node::ObjectWrap {
       Nan::SetPrototypeMethod(tpl, "digest", Digest);
 
       Nan::SetMethod(tpl, "hash", StaticHash);
-      target->Set(name, tpl->GetFunction());
+      Nan::Set(target, name, Nan::GetFunction(tpl).ToLocalChecked());
 
-      constructor_64.Reset(tpl);
+      constructor().Reset(Nan::GetFunction(tpl).ToLocalChecked());
```
Comment (mscdex) [inline src/hash_64.hpp]: I think we can get rid of this entirely.

File: src/hash_32.hpp [hunk 5]
```diff
@@ -158,9 +160,9 @@ class Hash32 : public node::ObjectWrap {
       Nan::SetPrototypeMethod(tpl, "update", Update);
       Nan::SetPrototypeMethod(tpl, "digest", Digest);
       Nan::SetMethod(tpl, "hash", StaticHash);
-      target->Set(name, tpl->GetFunction());
+      Nan::Set(target, name, Nan::GetFunction(tpl).ToLocalChecked());
 
-      constructor_32.Reset(tpl);
+      constructor().Reset(Nan::GetFunction(tpl).ToLocalChecked());
```
Comment (mscdex) [inline src/hash_32.hpp]: I think we can get rid of this entirely.

**Comment by asilvas** [comment]
Would be nice to move this PR forward.

**Comment by imrehg** [comment]
Would love to have this PR in, getting ready for Node 12 in a bunch of projects which use this :)

**Comment by asilvas** [comment]
@ben-page can you make the last requested changes? Not sure if that is what is holding this up..

**Comment by alexandrebodin** [comment]
Hi @mscdex @ben-page, any one of you willing to finish this PR ? and move forward ? Thanks !

File: src/hash_64.hpp [hunk 6]
```diff
@@ -160,9 +158,13 @@ class Hash64 : public node::ObjectWrap {
       else
         info.GetReturnValue().Set(convert_result(result, node::BUFFER));
     }
+	
+	static inline Nan::Persistent<v8::Function> & constructor() {
```
Comment (mscdex) [inline src/hash_64.hpp]: This function isn't needed. Same in hash_32.hpp

**Comment by mscdex** [comment]
Landed in 0609e6235e7fdc49b83b2754c779cb42858f668b, bd134d81d584edc83bcba83032e75e91d7d23cdd, and 6fa15b74508bd8fece248a6c4a0fe7e1229ceefc. Thanks!

**Pull request closed without merging**